doc/introduction/chemistry.rst

```diff
@@ -86,66 +86,63 @@ coordinates of each atomic species.
-multiplicity <https://en.wikipedia.org/wiki/Multiplicity_(chemistry)>`_ and the  `atomic basis
-functions <https://www.basissetexchange.org/>`_.
+For this, it is required to specify a string to label the molecule. Furthermore, the net charge,
+the `spin multiplicity <https://en.wikipedia.org/wiki/Multiplicity_(chemistry)>`_ and the  `atomic basis functions <https://www.basissetexchange.org/>`_ can also be specified.
 
 .. code-block:: python
 
     geometry = qml.qchem.read_structure('h2o.SDF')
-    hf_data = qml.qchem.meanfield_data(
+    hf_file = qml.qchem.meanfield(
         'water',
         geometry,
         charge=0,
-        multiplicity=1,
-        basis_set='sto-3g',
-        qc_package='pyscf'
+        mult=1,
+        basis='sto-3g',
+        package='pyscf'
     )
 
-The output variable ``hf_data`` stores the path to the directory containing the file ``'water.hd5'``
-with the Hartree-Fock electronic structure of the water molecule.
+The output variable ``hf_file`` stores the full path to the file ``'water.hd5'`` containing
+the Hartree-Fock electronic structure of the water molecule.
 
 Mapping the Hamiltonian to the Pauli basis
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The function :func:`~.active_space` is used to create an `active space <https://en.wikipedia
-.org/wiki/Complete_active_space>`__  by classifying the Hartree-Fock molecular orbitals as doubly-occupied,
-active, and external orbitals. Within this approximation, a certain number of *active electrons*
-can populate the *active orbitals*.
+.org/wiki/Complete_active_space>`__  by classifying the Hartree-Fock molecular orbitals as
+core, active, and external orbitals. Within this approximation, a certain number of *active electrons* can populate the *active orbitals*.
 
 .. code-block:: python
 
-    d_occ_indices, active_indices = qml.qchem.active_space(
-        'water',
-        hf_data,
-        n_active_electrons=2,
-        n_active_orbitals=2
+    water = MolecularData(filename=hf_file)
+    core, active = qml.qchem.active_space(
+        water.n_electrons,
+        water.n_orbitals,
+        nact_els=2,
+        nact_orbs=2
     )
 
-Once we have defined the active space, :func:`~.decompose_hamiltonian` calls
+Once we have defined the active space, :func:`~.decompose_molecular_hamiltonian` calls
 OpenFermion to generate the second-quantized fermionic Hamiltonian
 and map it to a linear combination of Pauli operators via the `Jordan-Wigner
 <https://en.wikipedia.org/wiki/Jordan%E2%80%93Wigner_transformation>`__ or `Bravyi-Kitaev
 <https://arxiv.org/abs/1208.5986>`__ transformation. For example,
 
 .. code-block:: python
 
-    qubit_hamiltonian = qml.qchem.decompose_hamiltonian(
-        'water',
-        hf_data,
+    qubit_hamiltonian = qml.qchem.decompose_molecular_hamiltonian(
+        hf_file,
         mapping='jordan_wigner',
-        docc_mo_indices=d_occ_indices,
-        active_mo_indices=active_indices
+        core=core,
```

Comment:
This is something I'm not a fan of, using `variable=variable`. This seems to arise from the fact that `core` and `active` are keyword arguments in the function, but I don't see a reason why this should be the case. If we remove this, we can simply pass `core` and `active` directly.

Reply:
Are `core` and `active` required arguments or optional arguments? If they are required, they should be positional, but if they are optional, best practice to keep them as keyword arguments in the signature.

Reply:
I already discussed this with @ixfoduap

Reply:
What was the solution?

Reply:
keeping the keywords